app/models/authenticator.rb

```diff
@@ -34,9 +34,23 @@ def uses_stored_password?
       false
     end
 
+    def can_authorize_user_by_userid?
```

Comment:
Not sure I like the `can` prefix.  How about just `def authorize_user_by_userid?`?